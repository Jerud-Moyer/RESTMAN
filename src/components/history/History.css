.History {
  display: flex;
  flex-direction: columns;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  border: blue 4px solid;
  border-radius: 8px;

  h1 {
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: large;
    text-shadow: 1px 1px 1px blue;
  }

  .item {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
  }

}
