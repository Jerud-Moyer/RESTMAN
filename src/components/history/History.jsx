import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './History.css';

const History = ({ searches }) => {

  const historyElements = searches.map(item => (
    <li key={ item.url }>
      <HistoryItem {...item} />
    </li>
  ));

  return (
    <div className={styles.History}>
      <h1>History</h1>
      <ul className={styles.Item}>
        {historyElements}
      </ul>
    </div>
  );

};

History.propTypes = {
  searches: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default History;
