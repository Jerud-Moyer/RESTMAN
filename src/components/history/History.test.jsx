import React from 'react';
import { render, cleanup } from '@testing-library/react';
import History from './History';

describe('History component', () => {
  afterEach(() => cleanup());

  it('renders History', () => {
    const { asFragment } = render(<History
      searches={[
        {
          method: 'GET',
          url: 'url.com'
        },
        {
          method: 'PUT',
          url: 'helloworld.com'
        },
        { 
          method: 'DELETE',
          url: 'immaprogrammer.com'
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
