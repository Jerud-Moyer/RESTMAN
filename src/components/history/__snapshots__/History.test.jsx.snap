// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<DocumentFragment>
  <div
    class="History"
  >
    <h1>
      History
    </h1>
    <ul
      class="Item"
    >
      <li>
        <div
          class="HistoryItem"
        >
          <span>
            GET
          </span>
          <span>
            url.com
          </span>
        </div>
      </li>
      <li>
        <div
          class="HistoryItem"
        >
          <span>
            PUT
          </span>
          <span>
            helloworld.com
          </span>
        </div>
      </li>
      <li>
        <div
          class="HistoryItem"
        >
          <span>
            DELETE
          </span>
          <span>
            immaprogrammer.com
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
